pkg/model/field.go

```diff
@@ -58,7 +57,14 @@ func newField(
 	if shapeRef != nil {
 		shape = shapeRef.Shape
 	}
-	if shape != nil {
+
+	if cfg != nil && cfg.IsSecret {
+		gt = "*acktypes.SecretKeyReference"
+		gte = "acktypes.SecretKeyReference"
+		gtwp = "*acktypes.SecretKeyReference"
+
+		crd.AddTypeImport("github.com/aws-controllers-k8s/runtime/pkg/types", "acktypes")
```

Comment:
No need for the above once you change to `ackv1alpha1`. See here:

https://github.com/aws-controllers-k8s/code-generator/blob/fe3ed9d62dbca1230c3850d8149c8da7fe15eddb/templates/apis/crd.go.tpl#L12